From d4e36951c4c2e93678c49190ec82cd3651d413f4 Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Thu, 14 May 2015 13:44:50 +0800
Subject: [PATCH 395/462] ov8858: Implement to get exposure time

This patch return ov8858 exposure time to user.

Change-Id: I555c824c144e5198b774def6a61e4c06f971904c
Tracked-On: #JIMINAN-36154
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/55930
Reviewed-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Build: IMIN (SC)
Reviewed-by: Soren Friis <soren.friis@intel.com>
Approver: Jian Xu Zheng <jian.xu.zheng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 6196108..9c7bbb4 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -1855,6 +1855,9 @@ static int ov8858_g_ctrl(struct v4l2_ctrl *ctrl)
 	case V4L2_CID_VFLIP:
 		ctrl->val = dev->vflip;
 		break;
+	case V4L2_CID_EXPOSURE_ABSOLUTE:
+		ctrl->val = dev->exposure;
+		break;
 	default:
 		dev_warn(&client->dev,
 			 "%s: Error: Invalid ctrl: 0x%X\n", __func__, ctrl->id);
@@ -2083,7 +2086,7 @@ static const struct v4l2_ctrl_config ctrls[] = {
 		.min = 0x0,
 		.step = 1,
 		.def = 0x00,
-		.flags = 0,
+		.flags = V4L2_CTRL_FLAG_READ_ONLY | V4L2_CTRL_FLAG_VOLATILE,
 	}, {
 		.ops = &ctrl_ops,
 		.id = V4L2_CID_TEST_PATTERN,
-- 
1.9.1

